In [7]:
import pandas as pd
spain_df = pd.read_csv('../data/spain.csv')
spain_df

Date  Season                home             visitor   HT   FT  \
0      1929-02-10    1928     Arenas de Getxo     Atletico Madrid  0-2  2-3   
1      1929-02-10    1928  Espanyol Barcelona          Real Union  1-0  3-2   
2      1929-02-10    1928         Real Madrid           CE Europa  0-0  5-0   
3      1929-02-10    1928       Real Sociedad     Athletic Bilbao  1-1  1-1   
4      1929-02-12    1928    Racing Santander        FC Barcelona  0-0  0-2   
...           ...     ...                 ...                 ...  ...  ...   
26190  2022-05-22    2021          Granada CF  Espanyol Barcelona  NaN  0-0   
26191  2022-05-22    2021          CA Osasuna        RCD Mallorca  NaN  0-2   
26192  2022-05-22    2021        FC Barcelona       Villarreal CF  NaN  0-2   
26193  2022-05-22    2021          Sevilla FC     Athletic Bilbao  NaN  1-0   
26194  2022-05-22    2021       Real Sociedad     Atletico Madrid  NaN  1-2   

       hgoal  vgoal  tier   round group notes  
0          2      3     1  league   NaN   NaN  
1          3      2     1  league   NaN   NaN  
2          5      0     1  league   NaN   NaN  
3          1      1     1  league   NaN   NaN  
4          0      2     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26190      0      0     1  league   NaN   NaN  
26191      0      2     1  league   NaN   NaN  
26192      0      2     1  league   NaN   NaN  
26193      1      0     1  league   NaN   NaN  
26194      1      2     1  league   NaN   NaN  

[26195 rows x 12 columns]

* 'Season' Questions: 
    - How many times did the length of the season change?
    - How many games did each team play per season?
    - Do the seasons only get longer?
    
    

How many times did the length of season change?

In [18]:
season_list = spain_df['Season'].unique()
spain_df['Season'].value_counts().unique()

array([462, 396, 380, 306, 305, 240, 182, 132,  90], dtype=int64)

This seems to suggest that the length of season has changed at least 8 times. The shortest season has 90 games and the longest has 462 games. It's possible that the season has changed back and forth between two of these values and would not be shown in this list.

-

How many games did each team play per season?

In [20]:
spain_df['Season'].unique()
home_team_list = spain_df['home'].unique()

array(['Arenas de Getxo', 'Espanyol Barcelona', 'Real Madrid',
       'Real Sociedad', 'Racing Santander', 'FC Barcelona',
       'Athletic Bilbao', 'Atletico Madrid', 'Real Union', 'CE Europa',
       'CD Alaves', 'Valencia CF', 'Real Betis', 'Real Oviedo',
       'Sevilla FC', 'Hercules CF', 'CA Osasuna', 'Real Zaragoza',
       'Celta Vigo', 'Real Murcia', 'Granada CF', 'Deportivo La Coruna',
       'CD Castellon', 'CE Sabadell', 'Sporting Gijon', 'CD Alcoyano',
       'Gimnastic', 'Real Valladolid', 'CD Malaga', 'UE Lleida',
       'Atletico Tetuan', 'UD Las Palmas', 'Real Jaen', 'CD Leonesa',
       'CD Condal', 'Elche CF', 'RCD Mallorca', 'CD Tenerife',
       'Cordoba CF', 'Pontevedra CF', 'Levante UD', 'Burgos CF',
       'UD Salamanca', 'Cadiz CF', 'Rayo Vallecano', 'Recreativo Huelva',
       'AD Almeria', 'CD Logrones', 'Real Burgos', 'Albacete',
       'SD Compostela', 'CP Merida', 'CF Extremadura', 'Villarreal CF',
       'CD Numancia', 'Malaga CF', 'Getafe CF', 'UD Almeri

Do the seasons only get longer?